core/src/main/java/org/apache/iceberg/SnapshotSummary.java

```diff
@@ -148,7 +148,7 @@ public void set(String property, String value) {
     }
 
     private void updatePartitions(PartitionSpec spec, ContentFile<?> file, boolean isAddition) {
-      if (trustPartitionMetrics) {
+      if (spec.isPartitioned() && trustPartitionMetrics) {
```

Comment:
How does this handle void partitions?

Reply:
In v1 once a partition field is dropped, the table is considered unpartitioned. The test [here](https://github.com/apache/iceberg/blob/d350c9b8c995a2953aa8b80a0a1fc7cadc4dd16a/core/src/test/java/org/apache/iceberg/TestPartitioning.java#L66-L86) shows this behavior. The `isPartitioned()` handles this case and was also updated by the same PR: https://github.com/apache/iceberg/pull/3059.